service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/ClientInstallationServiceImpl.java

```diff
@@ -83,10 +83,17 @@ public void addInstallation(Variant variant, Installation entity) {
 
     private void mergeCategories(Installation entity) {
         if (entity.getCategories() != null) {
-            List<String> categoryNames = convertToNames(entity.getCategories());
-            final List<Category> categories = categoryDao.findByNames(categoryNames);
-            entity.getCategories().removeAll(categories);
-            entity.getCategories().addAll(categories);
+
+            if(!entity.getCategories().isEmpty()) {
+                final List<String> categoryNames = convertToNames(entity.getCategories());
+                final List<Category> categories = categoryDao.findByNames(categoryNames);
+                entity.getCategories().removeAll(categories);
+                entity.getCategories().addAll(categories);
```

Comment:
so the deserialised categories never have an ID so we find all categories that already exist in the database and remove those and add them ( e.g. replace them ) leaving possible new categories that also don't have an ID yet.